### ISSUE TITLE
Different search paths for different subdirectories

### ISSUE THREAD

**Issue by shadowndacorner** [shader-slang/slang-vscode-extension#7]
Currently, the additionalSearchPaths configuration variable applies to any Slang source file in the workspace, however that doesn't necessarily map well to more complex build configurations.  It would be very nice to be able to specify that additional search paths only affect certain subdirectories.  For example, in an actual build, source files compiled under `{workspaceDir}/packages/shaders/pbr` may be able to see `{workspaceDir}/packages/shaders/shadowutils`, but not `{workspaceDir}/packages/shaders/lambert`.

I imagine this would be a new configuration variable that contains an array of something like `{ path: string, searchPaths: string[] }`.

**Comment by bmillsNV** [comment]
Likely an issue with Visual Studio Code. We might be able to provide an WAR for this, but don't have capacity right now. However, we welcome contributions if you'd like to see this fixed sooner.

**Comment by shadowndacorner** [comment]
@bmillsNV Is this already supported in the language server?  If so, if you could either point me in the general direction of how that is configured OR documentation on the language server in general (as, unless I'm missing it, it seems to be fairly undocumented), I'd be happy to look into implementing this as I described.  Given the available configuration options, I assumed that it wasn't currently supported in the language server, or would require multiple instances of the language server to support, which seems unnecessarily wasteful for large projects.

**Comment by jkwak-work** [comment]
I don't think we have any documented information about how to support this type of feature.

It had been a while since asked this question. But if you are still interested in, please i.plement and upstream to us.

Thanks.
